internal/redirecttraffic/redirect_traffic.go

```diff
@@ -67,19 +66,22 @@ func WithIPTablesProvider(provider iptables.Provider) TrafficRedirectionOpts {
 	}
 }
 
-func New(cfg *config.Config, proxySvc *api.AgentService, consulServerAddress, clusterARN string, proxyHealthCheckPort int, opts ...TrafficRedirectionOpts) TrafficRedirectionProvider {
+func New(cfg *config.Config, proxySvc *api.AgentService, consulServerAddress, clusterARN string, additionalInboundPortsToExclude []int, opts ...TrafficRedirectionOpts) TrafficRedirectionProvider {
 	trafficRedirectionCfg := &TrafficRedirectionCfg{
 		ProxySvc:             proxySvc,
 		ConsulServerAddress:  consulServerAddress,
 		ClusterARN:           clusterARN,
-		ProxyHealthCheckPort: proxyHealthCheckPort,
 		EnableConsulDNS:      cfg.ConsulDNSEnabled(),
 		ExcludeInboundPorts:  cfg.TransparentProxy.ExcludeInboundPorts,
 		ExcludeOutboundPorts: cfg.TransparentProxy.ExcludeOutboundPorts,
 		ExcludeOutboundCIDRs: cfg.TransparentProxy.ExcludeOutboundCIDRs,
 		ExcludeUIDs:          cfg.TransparentProxy.ExcludeUIDs,
 	}
 
+	if len(additionalInboundPortsToExclude) > 0 {
+		trafficRedirectionCfg.ExcludeInboundPorts = append(trafficRedirectionCfg.ExcludeInboundPorts, additionalInboundPortsToExclude...)
+	}
```

Comment:
minor nit: You don't need to check the length here. You can simply append `additionalInboundPortsToExclude`. If it is `nil` or empty the `append` is a no-op.

```suggestion
	trafficRedirectionCfg.ExcludeInboundPorts = append(trafficRedirectionCfg.ExcludeInboundPorts, additionalInboundPortsToExclude...)
```